### PR TITLE
[CPM-65] Create Course schema

### PR DESCRIPTION
**Pull request by HongShi10**
Created swagger documentation for the models and linked it to routes
Created Course Schema with tests
Created Regulation Schema with tests

### CODE REVIEW

**Review by Richardng12 — APPROVED** [review]
lgtm

File: api/src/models/course.js [hunk 1]
```diff
@@ -1,8 +1,98 @@
...
+    courseCode: {
+        type: String,
+        required: true,
+        unique: true,
+    },
+    points: {
+        type: Number,
+        required: true,
+    },
+    sem: {
```
Comment (Richardng12) [inline api/src/models/course.js:67]: Should be semester to be consistent with data model

Comment (HongShi10) [reply api/src/models/course.js]: ok

**Review by wryao64 — APPROVED** [review]
LGTM

**Pull request merged**